### PR TITLE
chore: Migrate FileUtilsTest.kt, NdkConfiguratorUtilsTest.kt and AgpConfiguratorUtilsTest.kt to AssertJ

### PR DESCRIPTION
**Pull request by henriquecicero**
## Summary:

Issue: [#45596](https://github.com/facebook/react-native/issues/45596)

## Changelog:

[INTERNAL] [CHANGED] - Migrated to AssertJ within files FileUtilsTest.kt, NdkConfiguratorUtilsTest.kt and AgpConfiguratorUtilsTest.kt

## Test Plan:

Run `./gradlew -p packages/gradle-plugin test`

### CODE REVIEW

**Comment by henriquecicero** [comment]
Hi, @cortinico, it's done.

**Comment by cortinico** [comment]
@henriquecicero can you please sign our CLA otherwise we won't be able to accept your contribution

**Comment by henriquecicero** [comment]
@cortinico, I did it just now.

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/FileUtilsTest.kt [hunk 1]
```diff
@@ -25,8 +25,8 @@ class FileUtilsTest {
 
     fileToMove.moveTo(destFile)
 
-    assertEquals("42", destFile.readText())
-    assertFalse(fileToMove.exists())
+    assertThat(destFile.readText()).isEqualTo("42")
+    assertThat(fileToMove.exists()).isFalse()
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/FileUtilsTest.kt:29]: Can you use:

```
assertThat(destFile).hasContent("42")
assertThat(fileToMove).exists()
```

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/FileUtilsTest.kt [hunk 2]
```diff
@@ -39,7 +39,7 @@ class FileUtilsTest {
 
     subFolder.recreateDir()
 
-    assertTrue(subFolder.exists())
-    assertEquals(0, subFolder.listFiles()?.size)
+    assertThat(subFolder).exists()
+    assertThat(subFolder.listFiles()).hasSize(0)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/FileUtilsTest.kt:43]: same here

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/NdkConfiguratorUtilsTest.kt [hunk 3]
```diff
@@ -18,29 +17,29 @@ class NdkConfiguratorUtilsTest {
   fun getPackagingOptionsForVariant_withHermesEnabled() {
     val (excludes, includes) = getPackagingOptionsForVariant(hermesEnabled = true)
 
-    assertTrue("**/libjsc.so" in excludes)
-    assertTrue("**/libjscexecutor.so" in excludes)
-    assertFalse("**/libjsc.so" in includes)
-    assertFalse("**/libjscexecutor.so" in includes)
+    assertThat("**/libjsc.so" in excludes).isTrue()
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/NdkConfiguratorUtilsTest.kt:20]: Please refactor this to use assertions like:
```
assertThat(excludes).containsExactly("**/libjsc.so", "**/libjscexecutor.so")
```

**Comment by henriquecicero** [comment]
@cortinico, thanks for the suggestions.

**Pull request closed without merging**